### PR TITLE
refactor(example): enable immutability checks

### LINKED ISSUE #1683: Example: enable immutability checks
With #1613 merged, we have to enable the `strictImmutability` check by providing `runtimeChecks` at https://github.com/ngrx/platform/blob/master/projects/example-app/src/app/app.module.ts#L38

This will act as a replacement of ngrx-store-freeze, which is already removed from the example-app.

If you start with this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by King-i**
add runtimeChecks to store root module
add runtimeChecks to tests, removes warning
set ActionReducerMap action type to any inline with rest of example to remove type issue

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #1683 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/example-app/src/app/auth/reducers/index.ts [hunk 1]
```diff
@@ -17,10 +16,7 @@ export interface State extends fromRoot.State {
   auth: AuthState;
 }
 
-export const reducers: ActionReducerMap<
-  AuthState,
-  AuthApiActions.AuthApiActionsUnion
-> = {
+export const reducers: ActionReducerMap<AuthState, any> = {
```
Comment (timdeschryver) [inline projects/example-app/src/app/auth/reducers/index.ts]: What's the reason of this change?

Comment (King-i) [reply projects/example-app/src/app/auth/reducers/index.ts]: Fixes typescript warning error due to action union, i believe its the reason the other `ActionReducerMap`'s are declared this way within the example app

Comment (timdeschryver) [reply projects/example-app/src/app/auth/reducers/index.ts]: Yea, you're right.
I'm just not seeing these errors in this file 😅
